sdks/python/apache_beam/utils/profiler.py

```diff
@@ -67,3 +67,77 @@ def __exit__(self, *args):
           self.profile, stream=s).sort_stats(Profile.SORTBY)
       self.stats.print_stats()
       logging.info('Profiler data: [%s]', s.getvalue())
+
+
+class MemoryReporter(object):
+  """A memory reporter that reports the memory usage and heap profile.
+  Usage:
+    mr = MemoryReporter(interval_sec=30.0)
+    mr.start()
+    while ...
+      <do something>
+      # this will report continuously with 30 seconds between reports.
+    mr.stop()
+
+  Or simply:
+    with MemoryReporter(interval_sec=100):
+      <do some thing>
+
+  Also it could report on demand without continuous reporting.
+    mr = MemoryReporter()  # default interval 60s but not activated.
+    <do something>
+    mr.report_once()
+  """
+
+  def __init__(self, interval_sec=60.0):
+    # guppy might not have installed. Set up the reporter only when guppy is
+    # installed.  http://pypi.python.org/pypi/guppy/0.1.10
+    try:
+      from guppy import hpy  # pylint: disable=import-error
+      self._hpy = hpy
+      self._interval_sec = interval_sec
+      self._enabled = False
+
+      def report_with_interval():
+        if not self._enabled:
+          return
+        self.report_once()
+        self._timer = Timer(self._interval_sec, report_with_interval)
+        self._timer.start()
+
+      self._timer = Timer(self._interval_sec, report_with_interval)
+
+    except ImportError:
+      logging.warning('guppy is not installed; MemoryReporter not available.')
+
+  def __enter__(self):
+    self.start()
+    return self
+
+  def __exit__(self, *args):
+    self.stop()
+
+  def start(self):
+    if self._enabled:
+      return
+    if not self._hpy:
+      logging.warning('guppy is not installed; MemoryReporter not available.')
```

Comment:
I think warning once in `__init__` is enough. This could be simplified to
```
if self._enabled or not self._hpy:
  return
```
Similarly for the `report_once`